src/main/java/com/spotify/futures/FuturesExtra.java

```diff
@@ -323,6 +474,23 @@ public Z apply(List<Object> results) {
     Z apply(A a, B b, C c, D d, E e, F f);
   }
 
+  /**
+   * Asynchronously transform the result of 3 {@link ListenableFuture}s a, b, c, d, e and f once
+   * both has completed using {@code function} and complete the returned ListenableFuture at the
+   * point.
+   * Execution is asynchronous in the sense that function returns a Future that may be executed
+   * asynchronously and once it completes it will complete the future that this method returns.
+   *
+   * @param a the first ListenableFuture
+   * @param b the second ListenableFuture
+   * @param c the third ListenableFuture
+   * @param d the fourth ListenableFuture
+   * @param e the fifth ListenableFuture
+   * @param f the sixth ListenableFuture
+   * @param function an AsyncFunction6 implementation, possibly expressed as a java 8 lambda.
+   * @return a ListenableFuture of the result of {@code function}, that completes as soon as
+   * the future that gets returned by the supplied Function6.apply() has completed.
```

Comment:
I'm not sure I follow. As far as I can tell, there is no way that the ListenableFuture<Z> returned by asyncTransform6 can complete without the ListenableFuture<Z> returned by the supplied AsyncFunction6 implementation has completed. Sure, I guess you can have an apply() that returns a ListenableFuture that completes immediately and then do a "transform" in a separate thread somewhere without the result of that transform being directly available in the code but would argue that that stretches the definition of transform a bit.

Or, as always, I could be totally confused, in which case I'm happy to be directed in the right direction :)